### PR TITLE
soundux: init at 0.2.7

### PR DESCRIPTION
**Pull request by D4ndellion**
###### Things done

All features work at least on pipewire, but I haven't tested them with pulseaudio.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
@GrahamcOfBorg eval

File: pkgs/applications/audio/soundux/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{
+  lib, stdenv, fetchFromGitHub, cmake, ninja, pkg-config, makeWrapper, wrapGAppsHook,
+  libappindicator-gtk3, openssl, pipewire, pulseaudio, webkitgtk, xorg,
+  libpulseaudio, libwnck3,
+  downloaderSupport ? true, ffmpeg, youtube-dl
+}:
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/soundux/default.nix:6]:
```suggestion
{ lib, stdenv, fetchFromGitHub, cmake, ninja, pkg-config, makeWrapper, wrapGAppsHook
, libappindicator-gtk3, openssl, pipewire, pulseaudio, webkitgtk, xorg
, libpulseaudio, libwnck3
, downloaderSupport ? true, ffmpeg, youtube-dl
}:
```

File: pkgs/applications/audio/soundux/default.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  installPhase = ''
+    mkdir -p $out/opt $out/bin
+    cp -r dist soundux-${version} $out/opt
+  '';
+
+  postFixup = ''
+    makeWrapper $out/opt/soundux-${version} $out/bin/soundux \
+      --prefix LD_LIBRARY_PATH ":" ${dynamicLibraries} \
+      "''${gappsWrapperArgs[@]}" \
+      ${lib.optionalString downloaderSupport "--prefix PATH \":\" " + downloaderPath}
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/soundux/default.nix:38]:
```suggestion
      ${lib.optionalString downloaderSupport "--prefix PATH \":\" " + (lib.makeBinPath [ffmpeg youtube-dl ])}
```

File: pkgs/applications/audio/soundux/default.nix [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  dontWrapGApps = true;
+
+  installPhase = ''
+    mkdir -p $out/opt $out/bin
+    cp -r dist soundux-${version} $out/opt
+  '';
+
+  postFixup = ''
+    makeWrapper $out/opt/soundux-${version} $out/bin/soundux \
+      --prefix LD_LIBRARY_PATH ":" ${dynamicLibraries} \
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/soundux/default.nix:36]:
```suggestion
      --prefix LD_LIBRARY_PATH ":" ${lib.makeLibraryPath [libpulseaudio pipewire libwnck3 ]} \
```

File: pkgs/applications/audio/soundux/default.nix [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+in
+stdenv.mkDerivation rec {
+  pname = "soundux";
+  version = "0.2.7";
+
+  src = fetchFromGitHub {
+    owner = "Soundux";
+    repo = "Soundux";
+    rev = version;
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/soundux/default.nix:21]: Do we have any of the deps there already in nixpkgs? If so they should be devondered.

Comment (dali99) [reply pkgs/applications/audio/soundux/default.nix:21]: What about the deps we don't have, should those also be devendored?

File: pkgs/applications/audio/soundux/default.nix (hunk 2 above)
Comment (dali99) [reply pkgs/applications/audio/soundux/default.nix:38]: this was on purpose

File: pkgs/applications/audio/soundux/default.nix (hunk 3 above)
Comment (dali99) [reply pkgs/applications/audio/soundux/default.nix:36]: this was on purpose

File: pkgs/applications/audio/soundux/default.nix (hunk 1 above)
Comment (dali99) [reply pkgs/applications/audio/soundux/default.nix:6]: I don't like this, is there any actual standard for this form? or is it your personal preference?

Comment (Mic92) [reply pkgs/applications/audio/soundux/default.nix:6]: Most code in nixpkgs follow this. nixpkgs-fmt will format the code in the same way.

Comment (SuperSandro2000) [reply pkgs/applications/audio/soundux/default.nix:6]: > I don't like this, is there any actual standard for this form?

This is the defacto standard and is what would be the standard if we would write it down. Please just follow it.

> or is it your personal preference?

No, it is actually not what I would personally do.

File: pkgs/applications/audio/soundux/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/audio/soundux/default.nix:38]: The variable is not reused anywhere so we don't actually need to declare it in the first place.

File: pkgs/applications/audio/soundux/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/applications/audio/soundux/default.nix:21]: It depends. If they are libs which can be packaged separate then they should be packaged. You can do that in this PR. If they are just a git repo with some files which are not used by anything else or don't really have a build system or are not indented to be packaged separate then not.

File: pkgs/applications/audio/soundux/default.nix (hunk 2 above)
Comment (dali99) [reply pkgs/applications/audio/soundux/default.nix:38]: So do we just not care about line-wrapping then?

Comment (SuperSandro2000) [reply pkgs/applications/audio/soundux/default.nix:38]: Not that strictly. It shouldn't be way to long but 100 or 120 is fine.

File: pkgs/applications/audio/soundux/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/audio/soundux/default.nix:6]: Also I think you need to replace libwnck3 with libwnck because the former is an alias.

**Comment by dali99** [comment]
I accidentally opened this PR with my master branch as the source, current package is available in the nur as `nur.repos.dandellion.soundux`

**Comment by weewoo22** [comment]
@dali99 Soundux isn't in NUR

**Comment by dali99** [comment]
Ah, yeah, my NUR has been broken for a while and I haven't checked why

I don't use Soundux anymore but here is the package I had when I was using it last:

```
{
  lib, stdenv, fetchFromGitHub, cmake, ninja, pkg-config, makeWrapper, wrapGAppsHook, makeDesktopItem, copyDesktopItems,
  libappindicator-gtk3, openssl, pipewire, pulseaudio, webkitgtk, xorg,
  libpulseaudio, libwnck3,
  downloaderSupport ? true, ffmpeg, youtube-dl
}:

let
  downloaderPath = lib.makeBinPath [ffmpeg youtube-dl ];
in
stdenv.mkDerivation rec {
  pname = "soundux";
  version = "0.2.7";

  src = fetchFromGitHub {
    owner = "Soundux";
    repo = "Soundux";
    rev = "0.2.7";
    fetchSubmodules = true;
    sha256 = "15kd9vl7inn8zm5cqzjkb6zb9xk2xxwpkm7fx1za3dy9m61sq839";
  };

  cmakeFlags = [ "-DCMAKE_BUILD_TYPE=Release" ];

  dontWrapGApps = true;

  installPhase = ''
    runHook preInstall

    mkdir -p $out/opt $out/bin
    cp -r dist soundux-${version} $out/opt

    runHook postInstall
  '';

  # Soundux loads pipewire, pulse and libwnck optionally during runtime
  postFixup = ''
    makeWrapper $out/opt/soundux-${version} $out/bin/soundux \
      --prefix LD_LIBRARY_PATH ":" ${lib.makeLibraryPath [libpulseaudio pipewire libwnck3 ]} \
      "''${gappsWrapperArgs[@]}" \
      ${lib.optionalString downloaderSupport "--prefix PATH \":\" " + downloaderPath}
  '';

  nativeBuildInputs = [ cmake ninja pkg-config makeWrapper wrapGAppsHook copyDesktopItems ];

  buildInputs = [
    libappindicator-gtk3
    openssl
    pipewire
    pulseaudio
    webkitgtk
    xorg.libX11
    xorg.libXtst
  ];

  desktopItems = makeDesktopItem {
    name = "Soundux";
    exec = pname;
    desktopName = "Soundux";
    genericName = "Soundboard";
    categories = "Audio;Music;Player;AudioVideo;";
    comment = "A universal soundboard that uses PulseAudio modules or PipeWire linking";
  };

  meta = with lib; {
    homepage = "https://soundux.rocks/";
    description = "cross-platform soundboard";
    license = licenses.gpl3Only;
    platforms = platforms.linux;
    maintainers = with maintainers; [ dandellion ];
  };
}
```

Now I just use my window manager for keybind to sound stuff. With some more custom pipewire setup

**Comment by Mic92** [comment]
Your repository in NUR probably does not evaluate and hence does not update.

**Pull request closed without merging**